moshi/src/main/java/com/squareup/moshi/JsonWriter.java

```diff
@@ -561,4 +565,30 @@ public final void endFlatten(int token) {
   public final String getPath() {
     return JsonScope.getPath(stackSize, scopes, pathNames, pathIndices);
   }
+
+  /** Returns the tag value for the given class key. */
+  @CheckReturnValue
+  public final @Nullable Object getTag(Class<?> clazz) {
+    if (tags == null) {
+      return null;
+    }
+    return tags.get(clazz);
+  }
+
+  /** Returns all tag values as an unmodifiable list. */
```

Comment:
I don’t think we want this method. You should need to know about a tag to see it! Otherwise tags become externally observable behavior

Reply:
sounds good. I'll remove it shortly.